### PR TITLE
fix:[CDS-54086]: Search connector by name

### PR DESCRIPTION
**Pull request by Fernando-Dourado**
## Describe your changes
- Using category to don't mix different types of settings stored in the same collection when searching by name.

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45388)

### CODE REVIEW

**Review by boopesh007 — DISMISSED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45388)* status: 4 file(s) to review. 1 of 2 approvals obtained.

File: 400-rest/src/main/java/software/wings/service/impl/SettingsServiceImpl.java [hunk 1]
```diff
@@ -1674,14 +1674,27 @@ public SettingAttribute getByName(String accountId, String appId, String attribu
...
+  @Override
+  public SettingAttribute getConnectorByName(String accountId, String appId, String attributeName) {
+    return getByNameAndCategory(accountId, appId, GLOBAL_ENV_ID, attributeName, SettingCategory.CONNECTOR);
+  }
+
+  private SettingAttribute getByNameAndCategory(
+      String accountId, String appId, String envId, String attributeName, SettingCategory category) {
+    final Query<SettingAttribute> query = wingsPersistence.createQuery(SettingAttribute.class)
+                                              .filter(SettingAttributeKeys.accountId, accountId)
+                                              .field("appId")
```
Comment (lasdox) [inline 400-rest/src/main/java/software/wings/service/impl/SettingsServiceImpl.java:1689]: Lets use constants for this
"SettingAttributeKeys.appId"

**Review by Fernando-Dourado — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45388)* status: 5 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/SettingsServiceImpl.java` line 1689 at r1](https://reviewable.io/reviews/harness/harness-core/45388#-NR05f5TXtlEVqhirz3r-r1-1689:-NR3H09R7ullchbkRJlS:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7e70ac333732f90a96ce652f5ab3ba2bf9a497ac/400-rest/src/main/java/software/wings/service/impl/SettingsServiceImpl.java#L1689)):*
<details><summary><i>Previously, lasdox (Lucas Sales) wrote…</i></summary><blockquote>

Lets use constants for this
"SettingAttributeKeys.appId"
</blockquote></details>

Done.
___
*[`400-rest/src/main/java/software/wings/service/impl/SettingsServiceImpl.java` line 1691 at r1](https://reviewable.io/reviews/harness/harness-core/45388#-NR05f5TXtlEVqhirz3r-r1-1691:-NR3H0laBSYLAwPDOGaL:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7e70ac333732f90a96ce652f5ab3ba2bf9a497ac/400-rest/src/main/java/software/wings/service/impl/SettingsServiceImpl.java#L1691)):*
<details><summary><i>Previously, lasdox (Lucas Sales) wrote…</i></summary><blockquote>

Lets use constants for this
"SettingAttributeKeys.envId"
</blockquote></details>

Done.

**Review by lasdox — APPROVED** [review]
> Reviewed 3 of 4 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45388)* status: 2. unresolved discussions.

**Review by lasdox — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45388)* status: :shipit: complete! ✓

**Pull request merged**